esx_service/vmdk_ops_test.py

```diff
@@ -252,7 +255,156 @@ def test_failure(self):
+                volName = 'VmdkAttachDetachTestVol' + str(id)
+                fullpath = os.path.join(path, volName + '.vmdk')
+                self.assertEqual(None,
+                                 vmdk_ops.createVMDK(vm_name='test-vm',
+                                                     vmdk_path=fullpath,
+                                                     vol_name=volName))
+                
+    def tearDown(self):
+        """ Cleanup after each test """
+        logging.debug("VMDKAttachDetachTest tearDown path")
+        self.cleanup()
+    
+    def create_vm(self, si, vm_name, datastore):
+        content = si.RetrieveContent()
+        datacenter = content.rootFolder.childEntity[0]
+        vmfolder = datacenter.vmFolder
+        hosts = datacenter.hostFolder.childEntity
+        resource_pool = hosts[0].resourcePool
+
+        datastore_path = '[' + datastore + '] ' + vm_name
+
+        # bare minimum VM shell, no disks. Feel free to edit
+        vmx_file = vim.vm.FileInfo(logDirectory=None,
+                                   snapshotDirectory=None,
+                                   suspendDirectory=None,
+                                   vmPathName=datastore_path)
+
+
+        config = vim.vm.ConfigSpec( 
+                                name=vm_name, 
+                                memoryMB=128, 
+                                numCPUs=1,
+                                files=vmx_file, 
+                                guestId='rhel5_64Guest', 
+                                version='vmx-11'
+                              )
+
+        #logging.debug("Createing VM %s", vm_name)
+        task = vmfolder.CreateVM_Task(config=config, pool=resource_pool)
+        vmdk_ops.wait_for_tasks(si, [task])
+
+        vm = [d for d in si.content.rootFolder.childEntity[0].vmFolder.childEntity if d.config.name == vm_name]
+        if (vm):
+            logging.debug("Found: VM %s", vm_name)
+            if vm[0].runtime.powerState == vim.VirtualMachinePowerState.poweredOff:
+                logging.debug("Attempting to power on %s", vm_name)
+                task = vm[0].PowerOnVM_Task()
+                vmdk_ops.wait_for_tasks(si, [task])
+    
+    def remove_vm(self, si, vm_name):
+
+        vm = [d for d in si.content.rootFolder.childEntity[0].vmFolder.childEntity if d.config.name == vm_name]
+        if (vm):
+            logging.debug("Found: VM %s", vm_name)
+            #logging.debug("The current powerState is  : %s", format(vm[0].runtime.powerState)))
+            if vm[0].runtime.powerState == vim.VirtualMachinePowerState.poweredOn:
+                logging.debug("Attempting to power off %s", vm_name)
+                task = vm[0].PowerOffVM_Task()
+                vmdk_ops.wait_for_tasks(si, [task])
+            
+            logging.debug("Trying to destroy VM %s", vm_name)    
+            task = vm[0].Destroy_Task()
+            vmdk_ops.wait_for_tasks(si, [task])
```

Comment:
For test purposes this can be done easily with vim-cmd as well. Basically create the VM then use "vim-cmd solo/register" to register the VM. The command will give a VM ID which can be used thereafter for any ops (power on/off, remove). Issue with using vmodl in this code is vmodl can change from release to release and the code needs to be updated to keep it running. The commands are stable.